### PR TITLE
added pilot-app with old vue app for mk portal

### PR DESCRIPTION
**Pull request by jeanpaulkommers**
Not production worthy code, but could be used as base

### CODE REVIEW

File: pilot-app/src/components/page/explorer.vue [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿<template>
+    <section class="component explorer" v-if="content">
+        <div class="widget" v-for="folder of content" :key="folder.id">
+            <h2>
+                <a :href="folder.folder.link.href" v-html="folder.folder.link.text"></a>
```
Comment (jeanpaulkommers) [inline pilot-app/src/components/page/explorer.vue:5]: Codacy has a fix for the issue: [Expected indentation of 8 spaces but found 16 spaces.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8170341)
```suggestion
            </h2>
```

File: pilot-app/src/components/page/mediakiwi-modal.vue [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        methods: {
+            closeAtOutsideClick() {
+                this.cancel();
+            },
+            cancel() {
+                this.close();
+            },
+            close() {
+                this.show = false
+                this.$emit('close')
```
Comment (jeanpaulkommers) [inline pilot-app/src/components/page/mediakiwi-modal.vue:59]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8170341)
```suggestion
                this.$emit"close"')
```

File: pilot-app/src/components/form/fields/wimChoiceRadio.vue [hunk 3]
```diff
@@ -0,0 +1,61 @@
+﻿<template>
+    <div :class="classname">        
+        <label v-if="prefix(field)" v-html="prefix(field)"></label>
+        <span v-for="option in field.options" :key="fieldID(option)">
+            <input type="radio"
+                   v-on="eventHandler(field, 'handleChange')"
+                   v-model="field.value"
+                   class="radio"
```
Comment (jeanpaulkommers) [inline pilot-app/src/components/form/fields/wimChoiceRadio.vue:8]: Codacy has a fix for the issue: [Expected indentation of 13 spaces but found 19 spaces.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8170341)
```suggestion
                   v-bind:class="errorClass(field)"
```

File: pilot-app/src/components/form/fields/wimTextArea.vue [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            }
+        },
+        methods: {
+            handleChange($event) {
+                console.log('triggerChange:textarea');
+                this.$emit('onchange', $event, this.field);
+            },
+            autoResize(element) {
+                element.style.height = 'auto';
+                element.style.height = element.scrollHeight + this.offset + 'px';
```
Comment (jeanpaulkommers) [inline pilot-app/src/components/form/fields/wimTextArea.vue:33]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8170341)
```suggestion
                element.style.height = element.scrollHeight + this.offset +"px"';
```

File: pilot-app/src/components/system/channel-selection.vue [hunk 5]
```diff
@@ -0,0 +1,47 @@
+﻿<template>
+    <div class="channel">
+        <h3>Channel Selection</h3>
+        <table>
+            <tbody>
+                <tr>
+                    <th>Channel Selection</th>
+                    <td>
+                        <wimChoiceDropdown :field="field" 
+                                           @onchange="change"></wimChoiceDropdown>
```
Comment (jeanpaulkommers) [inline pilot-app/src/components/system/channel-selection.vue:10]: Codacy has a fix for the issue: [Expected 1 line break before closing bracket, but no line breaks found.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8170341)
```suggestion
                                           @onchange="change
```

File: pilot-app/src/components/form/fields/wimChoiceCheckbox.vue [hunk 6]
```diff
@@ -0,0 +1,36 @@
+﻿<template>
+    <div :class="classname">
+        <label v-if="prefix(field)" v-html="prefix(field)"></label>
```
Comment (jeanpaulkommers) [inline pilot-app/src/components/form/fields/wimChoiceCheckbox.vue:3]: Codacy has a fix for the issue: [Expected indentation of 4 spaces but found 8 spaces.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8170341)
```suggestion
        <input type="checkbox"
```

File: pilot-app/src/components/form/wimForm.vue [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+                            :is="field.vueType"
+                            :field="field"
+                            v-model="field.value"
+                        ></component>
+                        <template v-else v-show="showField(field)">
+                            <th :class="expressCell(field.expression)" colspan="1">
+                                <label
+                                    v-if="!hideLabelForType(field.vueType)"
+                                    :for="field.propertyName"
+                                    :title="field.title"
```
Comment (jeanpaulkommers) [inline pilot-app/src/components/form/wimForm.vue:28]: Codacy has a fix for the issue: [Expected indentation of 18 spaces but found 36 spaces.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8170341)
```suggestion
                                    v-html="field.title"
```

File: pilot-app/src/components/form/wimForm.vue [hunk 8]
```diff
@@ -0,0 +1,316 @@
+﻿<template>
+    <section>
+        <table class="formTable">
```
Comment (jeanpaulkommers) [inline pilot-app/src/components/form/wimForm.vue:3]: Codacy has a fix for the issue: [Expected indentation of 4 spaces but found 8 spaces.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8170341)
```suggestion
            <tbody>
```

File: pilot-app/src/components/form/fields/wimTextline.vue [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+        mixins: [fieldMixin],
+        data() {
+            return {
+                value: "",
+            }
+        },
+        methods: {
+            getClass(field) {
+                let expression = '';//this.expressCell(field.expression);
+                if (typeof (field.className) !== 'undefined' && field.className) {
```
Comment (jeanpaulkommers) [inline pilot-app/src/components/form/fields/wimTextline.vue:25]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8170341)
```suggestion
                if (typeof (field.className) !=="undefined"' && field.className) {
```

File: pilot-app/src/components/form/fields/wimTreeselect.vue [hunk 10]
```diff
@@ -0,0 +1,68 @@
+﻿<template>
+    <div :class="fieldClass(field)" >
+        <label v-if="prefix(field)" v-html="prefix(field)"></label>
+        <treeselect v-model="field.value"
```
Comment (jeanpaulkommers) [inline pilot-app/src/components/form/fields/wimTreeselect.vue:4]: Codacy has a fix for the issue: ['v-model' should be on a new line.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8170341)
```suggestion
        <treeselec
```

File: pilot-app/src/output/plugin.js [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+    MediakiwiTextValue,
+    MediakiwiFormTableRow,
+    // Components
+    install: function (Vue, options) {
+        Vue.component('wimButton', wimButton);
+        Vue.component('wimChoiceCheckbox', wimChoiceCheckbox);
+        Vue.component('wimChoiceDropdown', wimChoiceDropdown);
+        Vue.component('wimChoiceRadio', wimChoiceRadio);
+        Vue.component('wimDateTime', wimDateTime);
+        Vue.component('wimDate', wimDate);
```
Comment (jeanpaulkommers) [inline pilot-app/src/output/plugin.js:64]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8170341)
```suggestion
        Vue.component"wimDate"', wimDate);
```

File: pilot-app/src/components/page/explorer.vue [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+            </h2>
+            <ul>
+                <li v-for="listItem in folder.listItems">
+                    <a :href="listItem.link.href" v-html="listItem.link.text" :title="listItem.link.text"></a>                    
+                </li>
+            </ul>
+        </div>
```
Comment (jeanpaulkommers) [inline pilot-app/src/components/page/explorer.vue:12]: Codacy has a fix for the issue: [Expected indentation of 4 spaces but found 8 spaces.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8170341)
```suggestion
    </section>
```

File: pilot-app/src/app.vue [hunk 13]
```diff
@@ -0,0 +1,156 @@
...
+                :sideNavigationContent="content.sideNavigation"
+                @onchange="onTopNavChange"
+            ></top-navigation>
+        </section>
+        <section class="bodyArticle">
+            <aside class="bodyAside">
+                <sideNavigation v-if="!isMobile && content && content.sideNavigation" :content="content.sideNavigation"></sideNavigation>
+            </aside>
+            <section class="bodyContent">
+                <listView ref="list" :IDs="pageIDs"></listView>
```
Comment (jeanpaulkommers) [inline pilot-app/src/app.vue:20]: Codacy has a fix for the issue: [Expected indentation of 8 spaces but found 16 spaces.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8170341)
```suggestion
            </section>
```

File: pilot-app/src/components/form/fields/wimTagVue.vue [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+                                text: option ? option.text : text,
+                                tiClasses: ["ti-valid"]
+                            };
+                        });
+                    }
+                    return [];
+                },
+                set(value) {
+                    // console.log(value);
+                    this.field.value = value.map(r => r.text);
```
Comment (jeanpaulkommers) [inline pilot-app/src/components/form/fields/wimTagVue.vue:62]: Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/Supershift/Sushi.Mediakiwi.UI/pullRequest?prid=8170341)
```suggestion
                    this.field.value = value.map(r)r => r.text);
```

**Review by lui297 — APPROVED** [review]
Looks good, we might need some better documentation of this later though!

**Pull request merged**